src/v/raft/consensus.cc

```diff
@@ -469,6 +469,7 @@ ss::future<result<replicate_result>> consensus::do_replicate(
                     // bound with last offset appended to the log
                     _visibility_upper_bound_index = std::max(
                       _visibility_upper_bound_index, res.last_offset);
+                    maybe_update_majority_replicated_index();
```

Comment:
But why is this here tho. The code is fine, but I am not sure why it is correct to have this here. The commit needs to add more information to understand. What is the flow that was missing. 

Why can't the loop advance the offsets always, why add this function here. Need more context.

Reply:
michal explained to me that we have 2 code paths. 

please udpate commit message.